src/spark_rapids_ml/knn.py

```diff
@@ -295,11 +277,89 @@ def _out_schema(self) -> Union[StructType, str]:
             ]
         )
 
-    def _get_cuml_fit_func(
+    def _require_ucx(self) -> bool:
+        """If enable or disable ucx over NCCL"""
+        return True
+
+    def _pre_process_data(  # type: ignore
+        self, dataset: DataFrame
+    ) -> Tuple[
+        List[Column], Optional[List[str]], int, Union[Type[FloatType], Type[DoubleType]]
+    ]:
+        (
+            select_cols,
+            multi_col_names,
+            dimension,
+            feature_type,
+        ) = super()._pre_process_data(dataset)
+
+        # if input format is vectorUDT, convert data type from float64 to float32
+        input_col, _ = self._get_input_columns()
+        if input_col is not None and isinstance(
+            dataset.schema[input_col].dataType, VectorUDT
+        ):
+            select_cols[0] = vector_to_array(col(input_col), dtype="float32").alias(
+                alias.data
+            )
+
+        select_cols.append(col(alias.label))
```

Comment:
Please make sure if the label type should be converted to be the same as features type.

Reply:
Yeah, it seems the current type is sufficient to differentiate query vectors from item vectors.